opentracing-util/src/main/java/io/opentracing/util/ThreadLocalScope.java

```diff
@@ -39,6 +44,8 @@
     public void close() {
         if (scopeManager.tlsScope.get() != this) {
             // This shouldn't happen if users call methods in the expected order. Bail out.
+            log.log(Level.SEVERE, "Scope being closed is not the active instance.",
```

Comment:
Personally I would log a warning at this point I think.
Also, wrap the log in an `if` for the level since you forcefully get a stacktrace.

Side note; My own code is a little more lenient with out-of-sequence closing; it won't touch the active scope, but it will 'mark' the wrong one closed. So when the correct one is closed, it will be 'skipped' in the stack unwinding code since it was already closed out of sequence.

I hope you get the idea, sorry it's still rather early in the morning here :wink: